Standardize CRootOf and RootSum with dedicated dummy symbols

vprayag2005 opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
see https://github.com/sympy/sympy/issues/26808
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->


#### Brief description of what is fixed or changed
This PR replaces the symbol inside CRootOf with a dummy symbol "_x". The problem is that xreplace gets confused by the symbol used inside CRootOf, as discussed in #28645.
In the printing code, the dummy symbol is replaced with the normal symbol "x" so that users can easily copy and paste the output.

This PR also applies the same approach to RootSum, using a dummy symbol "_w" internally and printing it as "w".

#### Other comments


#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers. Formerly, `log(-x)` incorrectly gave `-log(x)`.

* physics.units
  * Corrected a semantical error in the conversion between volt and statvolt which
    reported the volt as being larger than the statvolt.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
* polys
   * RootOf now uses an internal dummy symbol so that all instances will display using the symbol "x". This fixes some bugs where code involving RootOf would get confused if mixed with the symbol that was used to create the RootOf.
<!-- END RELEASE NOTES -->

Commit:
Replace symbol with dummy symbol(CRootOf)

Commit:
Merge remote-tracking branch 'upstream/master' into replace_symbol_with_dummy_crootof

Commit:
printing: use 'x' instead of dummy in CRootOf output

Commit:
Fix tests: Use symbol x instead of internal dummy _x

Commit:
Add tests for CRootOf xreplace and Poly construction

- test_CRootOf_xreplace: verify xreplace behavior with CRootOf
- test_Poly_with_CRootOf: verify Poly construction from RootOf

vprayag2005 commented on sympy/polys/rootoftools.py:
```diff
@@ -365,6 +366,7 @@ def _new(cls, poly, index):
         obj = Expr.__new__(cls)
 
+        poly = poly.replace(poly.gen, _rootof_dummy)
         obj.poly = PurePoly(poly)
```
> I think you have to replace PurePoly with Poly everywhere in this file.

Should I also change RootSum to use Poly, or only CRootOf? You mentioned 'everywhere,' so I wanted to clarify it.

Commit:
refactor: PurePoly -> Poly in rootoftools(CRootof) and update tests

Commit:
docs: Update _pure_key_dict doctest for Poly behavior

oscarbenjamin commented on sympy/polys/rootoftools.py:
```diff
@@ -1166,7 +1178,9 @@ def __new__(cls, expr, func=None, x=None, auto=True, quadratic=False):
         rational = cls._is_func_rational(poly, func)
-        factors, terms = _pure_factors(poly), []
+        _, factors = poly.factor_list()
+        factors = [(PurePoly(f, expand=False), m) for f, m in factors]
```
Why is this using PurePoly?

oscarbenjamin replied:
It might make sense to handle RootSum in a separate PR but ultimately we should not use PurePoly for RootSum either.

oscarbenjamin commented:
> I suspect this caused by a domain issue

What are the domains in that case?

That could also be caused by mismatching dummy symbols.

oscarbenjamin commented:
The Poly.retract method should be used to simplify the domain (if that is needed):
```
In [1]: p = Poly(z, x, domain=ZZ[y,z])

In [2]: p
Out[2]: Poly(z, x, domain='ZZ[y,z]')

In [3]: p.retract()
Out[3]: Poly(z, x, domain='ZZ[z]')
```

Commit:
RootSum: Use Poly instead of PurePoly with dummy symbol '_x'

Commit:
printing: Display RootSum with 'x' instead of '_x' dummy symbol

Commit:
Remove unused import

Commit:
Use dedicated _rootsum_dummy symbol for RootSum

Add module-level _rootsum_dummy = Dummy('w') for consistent representation.
Update RootSum to use _rootsum_dummy instead of _rootof_dummy.
Update printers and tests to reflect 'w' symbol.

Commit:
fix doctests
